src/gsIO/gsParaviewDataSet.cpp

```diff
@@ -0,0 +1,108 @@
+{
+    gsParaviewDataSet::gsParaviewDataSet(std::string basename,
+                    gsMultiPatch<real_t> * const geometry,
+                    gsExprEvaluator<real_t> * eval,
+                    gsOptionList options)
+                    :m_basename(basename),
+                    m_geometry(geometry),
+                    m_evaltr(eval),
+                    m_options(options)
+    {
+        unsigned nPts = m_options.askInt("numPoints",1000);
+
+        // QUESTION: Can I be certain that the ids are consecutive?
+        initFilenames();
+        for ( index_t k=0; k!=m_geometry->nPieces(); k++) // For every patch.
+        {
+            gsMatrix<real_t> activeBases = m_geometry->piece(k).support();
+            gsGridIterator<real_t,CUBE> pt(activeBases, nPts);
+
+            const gsVector<index_t> & np( pt.numPointsCwise() );
+            index_t np1 = (np.size()>1 ? np(1)-1 : 0);
+            index_t np2 = (np.size()>2 ? np(2)-1 : 0);
+
+            // initializes individual .vts files
+            // for every patch
+            std::ofstream file(m_filenames[k].c_str());
+            file << std::fixed; // no exponents
+            file << std::setprecision(5); // PLOT_PRECISION
+            file <<"<?xml version=\"1.0\"?>\n";
+            file <<"<VTKFile type=\"StructuredGrid\" version=\"0.1\">\n";
+            file <<"<StructuredGrid WholeExtent=\"0 "<< np(0)-1<<" 0 "<< np1 <<" 0 "
+                << np2 <<"\">\n";
+            file <<"<Piece Extent=\"0 "<< np(0)-1<<" 0 "<<np1<<" 0 "
+                << np2 <<"\">\n";
+            file <<"<PointData>\n";
+            file.close();
+        }
+    }
+
+    
+    const std::vector<std::string> gsParaviewDataSet::filenames()
+    {
+        return m_filenames;
+    }
+
+
+    void gsParaviewDataSet::save()
+    {
+        unsigned nPts = m_options.askInt("numPoints",1000);
+        unsigned precision = m_options.askInt("precision",5);
+        bool plotElements   = m_options.askSwitch("plotElements", false);
+        bool plotControlNet = m_options.askSwitch("plotControlNet", false);
+
+        std::vector<std::string> points = toVTK(*m_geometry,nPts,precision); //m_evaltr->geoMap2vtk(*m_geometry,nPts, precision);
+        // QUESTION: Can I be certain that the ids are consecutive?
+        for ( index_t k=0; k!=m_geometry->nPieces(); k++) // For every patch.
+        {
+            std::ofstream file;
+            file.open(m_filenames[k].c_str(), std::ios_base::app); // Append to file 
+            file <<"</PointData>\n\n\n<!-- GEOMETRY -->\n<Points>\n";
+            file << points[k];
+            file << "</Points>\n</Piece>\n</StructuredGrid>\n</VTKFile>";
+            file.close();
+            if (plotControlNet)
+            {
+                writeSingleControlNet( m_geometry->piece(k), m_basename + "_cnet" + std::to_string(k));
+                m_filenames.push_back( m_basename + "_cnet" + std::to_string(k)+".vtp");
+            } 
+            if ( plotElements)
```

Comment:
Here: there should be all the code from the gsWriteParaview function, in particular the computation of the resolution

Reply:
I don't quite get what you are suggesting, could you elaborate ?

Reply:
Implemented in [60f9deb](https://github.com/gismo/gismo/pull/572/commits/60f9deb05b79554f9487160d62320abcfc18b343) if you want to have a look